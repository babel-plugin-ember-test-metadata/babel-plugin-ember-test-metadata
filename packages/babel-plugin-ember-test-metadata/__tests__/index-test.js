const pluginTester = require('babel-plugin-tester').default;
const addMetadata = require('../index');
const path = require('path');

pluginTester({
  plugin: addMetadata,
  pluginName: 'addMetadata',
  snapshot: true,
  pluginOptions: {
    enabled: true,
    packageName: 'packages/babel-plugin-ember-test-metadata',
  },
  tests: [
    {
      title: 'for a module with hooks it adds getTestMetadata',
      fixture: path.join(__dirname, '__fixtures__', 'with-hooks-test.js'),
    },
    {
      title: 'for a module with no hooks it adds test metadata',
      fixture: path.join(__dirname, '__fixtures__', 'without-hooks-test.js'),
    },
    {
      title: 'with existing metadata import it reuses the import',
      fixture: path.join(__dirname, '__fixtures__', 'with-get-test-metadata-test.js'),
    },
    {
      title: 'with multiple sibling modules',
      fixture: path.join(__dirname, '__fixtures__', 'with-multiple-modules-test.js'),
    },
    {
      title: 'with no module callback it does not add a beforeEach',
      fixture: path.join(__dirname, '__fixtures__', 'with-no-module-callback-test.js'),
    },
    {
      title: 'Embroider build path',
      fixture: path.join(
        __dirname,
        '__fixtures__',
        'temp-test-123',
        'embroider',
        '123456',
        'tests',
        'embroider-prefix-test.js'
      ),
      pluginOptions: {
        enabled: true,
        packageName: 'packages/babel-plugin-ember-test-metadata',
        isUsingEmbroider: true,
      },
    },
  ],
});
