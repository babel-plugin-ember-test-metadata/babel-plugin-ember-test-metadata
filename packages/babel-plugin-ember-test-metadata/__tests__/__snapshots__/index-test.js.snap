// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addMetadata Embroider build path: Embroider build path 1`] = `

import { module, test } from 'qunit';
import { click, getTestMetadata } from '@ember/test-helpers';

module('Acceptance | example test embroider prefix', function (hooks) {
  hooks.beforeEach(function () {
    // noop
  });

  test('example', async function (assert) {
    click();
    assert(true);
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

import { module, test } from 'qunit';
import { click, getTestMetadata } from '@ember/test-helpers';
module('Acceptance | example test embroider prefix', function (hooks) {
  hooks.beforeEach(function () {
    let testMetadata = getTestMetadata(QUnit.config.current.testEnvironment);
    testMetadata.filePath = 'pathtest/embroider-prefix-test.js';
  });
  hooks.beforeEach(function () {
    // noop
  });
  test('example', async function (assert) {
    click();
    assert(true);
  });
});


`;

exports[`addMetadata for a module with hooks it adds getTestMetadata: for a module with hooks it adds getTestMetadata 1`] = `

import {module, test} from 'qunit';
import { click } from '@ember/test-helpers';

module('Acceptance | example test', function (hooks) {
  hooks.beforeEach(function () {
    // noop
  });

  test('example', async function (assert) {
    click();
    assert(true)
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

import { getTestMetadata } from '@ember/test-helpers';
import { module, test } from 'qunit';
import { click } from '@ember/test-helpers';
module('Acceptance | example test', function (hooks) {
  hooks.beforeEach(function () {
    let testMetadata = getTestMetadata(QUnit.config.current.testEnvironment);
    testMetadata.filePath = '__tests__/__fixtures__/with-hooks-test.js';
  });
  hooks.beforeEach(function () {
    // noop
  });
  test('example', async function (assert) {
    click();
    assert(true);
  });
});


`;

exports[`addMetadata for a module with no hooks it adds test metadata: for a module with no hooks it adds test metadata 1`] = `

import {module, test} from 'qunit';
import { click } from '@ember/test-helpers';

module('Acceptance | example test', function () {
  hooks.beforeEach(function () {
    // noop
  });

  test('example', async function (assert) {
    click();
    assert(true)
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

import { getTestMetadata } from '@ember/test-helpers';
import { module, test } from 'qunit';
import { click } from '@ember/test-helpers';
module('Acceptance | example test', function (hooks) {
  hooks.beforeEach(function () {
    let testMetadata = getTestMetadata(QUnit.config.current.testEnvironment);
    testMetadata.filePath = '__tests__/__fixtures__/without-hooks-test.js';
  });
  hooks.beforeEach(function () {
    // noop
  });
  test('example', async function (assert) {
    click();
    assert(true);
  });
});


`;

exports[`addMetadata with existing metadata import it reuses the import: with existing metadata import it reuses the import 1`] = `

import {module, test} from 'qunit';
import { click, getTestMetadata } from '@ember/test-helpers';

module('Acceptance | example test', function (hooks) {
  hooks.beforeEach(function () {
    // noop
  });

  test('example', async function (assert) {
    click();
    assert(true)
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

import { module, test } from 'qunit';
import { click, getTestMetadata } from '@ember/test-helpers';
module('Acceptance | example test', function (hooks) {
  hooks.beforeEach(function () {
    let testMetadata = getTestMetadata(QUnit.config.current.testEnvironment);
    testMetadata.filePath = '__tests__/__fixtures__/with-get-test-metadata-test.js';
  });
  hooks.beforeEach(function () {
    // noop
  });
  test('example', async function (assert) {
    click();
    assert(true);
  });
});


`;

exports[`addMetadata with multiple sibling modules: with multiple sibling modules 1`] = `

import {module, test} from 'qunit';
import { click } from '@ember/test-helpers';

module('Acceptance | example test', function (hooks) {
  hooks.beforeEach(function () {
    // noop
  });

  test('example', async function (assert) {
    click();
    assert(true)
  });
});

module('Acceptance | example test 2', function (hooks) {
  hooks.beforeEach(function () {
    // noop
  });

  test('example', async function (assert) {
    click();
    assert(true)
  });
});

      ↓ ↓ ↓ ↓ ↓ ↓

import { getTestMetadata } from '@ember/test-helpers';
import { module, test } from 'qunit';
import { click } from '@ember/test-helpers';
module('Acceptance | example test', function (hooks) {
  hooks.beforeEach(function () {
    let testMetadata = getTestMetadata(QUnit.config.current.testEnvironment);
    testMetadata.filePath = '__tests__/__fixtures__/with-multiple-modules-test.js';
  });
  hooks.beforeEach(function () {
    // noop
  });
  test('example', async function (assert) {
    click();
    assert(true);
  });
});
module('Acceptance | example test 2', function (hooks) {
  hooks.beforeEach(function () {
    let testMetadata = getTestMetadata(QUnit.config.current.testEnvironment);
    testMetadata.filePath = '__tests__/__fixtures__/with-multiple-modules-test.js';
  });
  hooks.beforeEach(function () {
    // noop
  });
  test('example', async function (assert) {
    click();
    assert(true);
  });
});


`;

exports[`addMetadata with no module callback it does not add a beforeEach: with no module callback it does not add a beforeEach 1`] = `

import {module} from 'qunit';

module('Acceptance | example test no module');

      ↓ ↓ ↓ ↓ ↓ ↓

import { getTestMetadata } from '@ember/test-helpers';
import { module } from 'qunit';
module('Acceptance | example test no module');


`;
